### PR TITLE
Encrypt and decrypt state from chaincode

### LINKED ISSUE #497: Encrypt and decrypt state from chaincode

### PR DESCRIPTION
**Pull request by binhn**
closes #497

### CODE REVIEW

**Comment by binhn** [comment]
Signed-off-by: binh nguyen binhn@us.ibm.com

**Comment by binhn** [comment]
tested with @muralisrini

File: core/ledger/kvledger/benchmark/chainmgmt/sanity_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestChainMgmt is a basic sanity check test to catch any errors that could be caused by changes in the ledgermgmt ot kvledger packages
+func TestChainMgmt(t *testing.T) {
+	dataDir := "/tmp/kvledger/benchmark/sanitycheck"
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/sanity_test.go]: Please use an actual temporary directory for this test.

File: core/ledger/kvledger/benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Ledger Benchmark Tests
+
+This readme explains the working and the usage of Benchmarks in this package.
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/README.md]:
```suggestion
This readme explains the Benchmarks in this package and how to run them.
```

File: core/ledger/kvledger/benchmark/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+- What is the Ledger Benchmark tests
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/README.md]:
```suggestion
- What are the Ledger Benchmark tests
```

File: core/ledger/kvledger/benchmark/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- How to Run the Tests
+
+## What is the Ledger Benchmark tests
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/README.md]:
```suggestion
## What are the Ledger Benchmark tests
```

File: core/ledger/kvledger/benchmark/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+The Ledger Benchmark tests provides a way to measure performance of the Ledger
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/README.md]:
```suggestion
The Ledger Benchmark tests provide a way to measure the performance of the Ledger
```

File: core/ledger/kvledger/benchmark/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+component with a flexible and controlled workload. These benchmarks short-circuit
+the chaincode and directly use transaction simulator. Also, for block creation,
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/README.md]:
```suggestion
chaincode go directly to the transaction simulator. Also, for block creation,
```

File: core/ledger/kvledger/benchmark/chainmgmt/chains.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/hyperledger/fabric/core/ledger/ledgermgmt"
+	"github.com/hyperledger/fabric/core/ledger/ledgermgmt/ledgermgmttest"
+	"github.com/hyperledger/fabric/core/ledger/util/couchdb"
+)
+
+// ChainID is a type used for the ids for the chains for experiments
+type ChainID int
+
+func (chainID ChainID) String() string {
+	return fmt.Sprintf("%s%04d", "chain-", chainID)
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/chains.go]:
```suggestion
	return fmt.Sprintf("chain-%04d", chainID)
```

**Pull request merged**